src/test/ui/issues/issue-49040.stderr

```diff
@@ -7,10 +7,8 @@ LL | #![allow(unused_variables)];
 error[E0601]: `main` function not found in crate `issue_49040`
   --> $DIR/issue-49040.rs:1:1
    |
-LL | / #![allow(unused_variables)];
-LL | |
-LL | | fn foo() {}
-   | |__^ consider adding a `main` function to `$DIR/issue-49040.rs`
+LL | #![allow(unused_variables)];
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ consider adding a `main` function to `$DIR/issue-49040.rs`
```

Comment:
Is this change intentional? There's not really a great span for this error, so I suppose an attribute is as good as any.

Reply:
Yes, there's a slight change in the crate span in case of parsing errors due to changes in `fn parse_crate_mod`.
Previously it went one token ahead into the unparsed part of the crate (see how `^` points to `fn`), now it covers the parsed part consistently with non-root modules.